File: scratch/fix_urls.py
import os

def replace_in_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    # First, fix the broken replacement from previous attempt
    new_content = content.replace("$/", "`${import.meta.env.VITE_API_URL}/`")
    # Also handle the original case if it wasn't broken
    new_content = new_content.replace("http://localhost:5000/api", "`${import.meta.env.VITE_API_URL}`")
    
    # Special case for strings already in backticks
    # If it's already `${import.meta.env.VITE_API_URL}`, we might have double backticks or nested.
    # Wait, the previous attempt replaced 'http://localhost:5000/api' with '$/'.
    # So if it was 'http://localhost:5000/api/auth/login', it became '$/auth/login'.
    # If the original was inside single quotes: axios.post('http://localhost:5000/api/auth/login')
    # It became axios.post('$/auth/login')
    
    # Let's be more precise.
    # We want axios.post(`${import.meta.env.VITE_API_URL}/auth/login`)
    
    if content != new_content:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(new_content)
        print(f"Updated {filepath}")

def process_dir(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(".jsx") or file.endswith(".js"):
                replace_in_file(os.path.join(root, file))

process_dir("admin-dashboard/src")
process_dir("frontend/src")

File: scratch/fix_urls.js
const fs = require('fs');
const path = require('path');

function replaceInFile(filepath) {
    let content = fs.readFileSync(filepath, 'utf8');
    
    // Fix the broken replacement from previous attempt: '$/'
    // But be careful not to replace it if it's already part of a template literal
    // Actually, I'll just look for '$/'
    let newContent = content.replace(/\'\$\/\'/g, "`${import.meta.env.VITE_API_URL}`"); // handle single quotes
    newContent = newContent.replace(/\'\$\//g, "`${import.meta.env.VITE_API_URL}/"); // handle prefix in single quotes
    
    // Replace original hardcoded URLs
    newContent = newContent.replace(/http:\/\/localhost:5000\/api/g, "${import.meta.env.VITE_API_URL}");
    
    // After replacing 'http://localhost:5000/api' with '${import.meta.env.VITE_API_URL}',
    // we need to make sure the enclosing quotes are backticks.
    
    // Regex to find '...${import.meta.env.VITE_API_URL}...' and change quotes to backticks
    // This is hard to do perfectly with regex, so I'll do a few common patterns.
    newContent = newContent.replace(/\'(\$\{import\.meta\.env\.VITE_API_URL\}[^\']*)\'/g, "`$1`");
    
    if (content !== newContent) {
        fs.writeFileSync(filepath, newContent, 'utf8');
        console.log(`Updated ${filepath}`);
    }
}

function processDir(directory) {
    const files = fs.readdirSync(directory);
    for (const file of files) {
        const fullPath = path.join(directory, file);
        const stat = fs.statSync(fullPath);
        if (stat.isDirectory()) {
            processDir(fullPath);
        } else if (file.endsWith('.jsx') || file.endsWith('.js')) {
            replaceInFile(fullPath);
        }
    }
}

processDir('admin-dashboard/src');
processDir('frontend/src');
